### ISSUE TITLE
[SYCL-MLIR] Use `VectorType` for different vector types

### ISSUE THREAD

**Issue by whitneywhtsang** [intel/llvm#7356]
In https://github.com/intel/llvm/pull/7342, `ext_vector_type` is handled, but not the other vector types.

https://github.com/intel/llvm/pull/7342#pullrequestreview-1175435255:
(see [here](https://clang.llvm.org/docs/LanguageExtensions.html#vectors-and-extended-vectors)). The following should have the same output regardless of whether we use `ext_vector_type(4)` or `vector_size(4 * sizeof(int))`:
```cpp
using vec = int __attribute__((vector_size(4 * sizeof(int))));

vec id(vec a) { return a; }
```

**Comment by etiotto** [comment]
@whitneywhtsang still valid ?

**Comment by whitneywhtsang** [comment]
Yes, doesn't need to be assign to me, whoever have time can look into this.
